src/daemon/screenlock.vala

```diff
@@ -0,0 +1,312 @@
+
+            if (idle_delay !=0 ) {
+                new_idle += "timeout " + idle_delay.to_string() + " 'wlopm --off \\*' resume 'wlopm --on \\*' ";
+            }
+
+            if (lock_enabled && !disable_lock_screen) {
+                new_idle += "timeout " + (idle_delay + lock_delay).to_string() + " '" + calculate_lockcommand() + "' ";
+            }
+
+            if (!disable_lock_screen) {
+                new_idle += "before-sleep '" + calculate_lockcommand() + "'";
+            }
+
+            new_idle += " " + calculate_sleep();
+
+            debug("%s", new_idle);
+            try {
+                string[] spawn_args = {"killall", "swayidle"};
+                string[] spawn_env = Environ.get();
+
+                /* sometimes swayidle exists twice - so we run the kill Process
+                   to make sure before starting swayidle again
+                */
+                for (var i = 0; i <= 1; i++) {
+                    Process.spawn_sync ("/",
+                                spawn_args,
+                                spawn_env,
+                                SpawnFlags.SEARCH_PATH |
+                                SpawnFlags.STDERR_TO_DEV_NULL |
+                                SpawnFlags.STDOUT_TO_DEV_NULL,
+                                null,
+                                null,
+                                null,
+                                null);
+                }
+
+                Process.spawn_command_line_async(new_idle);
+            } catch (SpawnError e) {
+				print("Error: %s\n", e.message);
+			}
+        }
+
+        [DBus (visible = false)]
+		public void setup_dbus() {
+			/* Hook up screenlock dbus */
+			Bus.own_name(BusType.SESSION, DBUS_SCREENLOCK, BusNameOwnerFlags.REPLACE,
+				on_bus_acquired,
+				() => {},
+				() => {} );
+		}
+
+        void on_bus_acquired(DBusConnection conn) {
+			try {
+				conn.register_object(DBUS_SCREENLOCK_PATH, this);
+			} catch (Error e) {
+				message("Unable to register Screenlock: %s", e.message);
+			}
+		}
+
+        public async void lock() throws GLib.DBusError, GLib.IOError {
+            if (!all_apps) {
+                yield;
+                return;
+            }
+
+            try {
+                Process.spawn_command_line_async(calculate_lockcommand());
+            } catch (SpawnError e) {
+				print("Error: %s\n", e.message);
+			}
+
+            yield;
+        }
+
+		[DBus (visible = false)]
+		public static unowned Screenlock init() {
+			if (instance == null)
+				instance = new Screenlock();
+
+			return instance;
+		}
+
+        public bool on_battery() throws GLib.Error {
+            return battery_mode;
+        }
+
+        private void client_daemon(GLib.Object obj, GLib.ParamSpec? sp) {
+            battery_mode = client.get_on_battery();
+        }
+
+        private Screenlock() {
+            string check_apps[] = {"swayidle", "killall", "wlopm", "systemctl", "dbus-send"};
```

Comment:
As above, if we use loginctl we won't be hard-depending on `systemctl` existing, just logind.